File: bdp-master-client/src/main/scala/com/github/bdp/master/client/service/AppService.scala
package com.github.bdp.master.client.service

import com.github.bdp.master.client.Constants._
import com.github.bdp.master.client.domain.App
import com.github.bdp.master.client.util.{JsonDecoder, RedisClient}
import com.typesafe.scalalogging.LazyLogging

object AppService extends LazyLogging {

  def getAppBy(id: Long): App = {
    JsonDecoder.decodeApp(RedisClient.get(s"$APP_KEYSPACE:$id"))
  }

  def getAppBy(name: String): App = {
    val key = RedisClient.get(s"i_$APP_KEYSPACE:$name")
    JsonDecoder.decodeApp(RedisClient.get(key))
  }

}

File: bdp-master-server/src/main/java/com/github/bdp/master/server/service/AppService.java
package com.github.bdp.master.server.service;


import com.github.bdp.master.server.domain.App;

import java.util.List;

public interface AppService {

	App findOne(Long id);

	App findOne(String appName);

	void save(App app);

	void delete(Long id);

	List<App> findAll();

	void loadAll();
}
